File: pdu/python/std_msgs/pdu_cdr_conv_Byte.py
from .pdu_pytype_Byte import Byte
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_Byte(writer: CdrWriter, src: Byte):
    writer.write_uint8(src.data)


def cdr_body_to_py_Byte(reader: CdrReader, dst: Byte):
    dst.data = reader.read_uint8()
    return dst


def py_to_cdr_Byte(src: Byte) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_Byte(writer, src)
    return writer.bytes()


def cdr_to_py_Byte(cdr_payload) -> Byte:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = Byte()
    return cdr_body_to_py_Byte(reader, dst)

File: pdu/javascript/std_msgs/pdu_cdr_conv_Byte.js
import { Byte } from './pdu_jstype_Byte.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduByteConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { Byte } src
     */
    static to_cdr_body(writer, src) {
        writer.write_uint8(src.data);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { Byte } dst
     * @returns { Byte }
     */
    static cdr_body_to_js(reader, dst) {
        dst.data = reader.read_uint8();
        return dst;
    }

    /**
     * @param { Byte } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { Byte }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new Byte());
    }
}
